tests/tools/oecertdump/host/host.cpp

```diff
@@ -81,97 +89,144 @@ oe_result_t enclave_identity_verifier(oe_identity_t* identity, void* arg)
 void output_certificate(const uint8_t* data, size_t data_len)
 {
 #if defined(__linux__)
-    if (log_file)
+    printf("\n");
+    X509* x509;
+    BIO* input = BIO_new_mem_buf(data, (int)data_len);
+    x509 = d2i_X509_bio(input, NULL);
+    if (x509)
     {
-        fprintf(log_file, "\n");
-        X509* x509;
-        BIO* input = BIO_new_mem_buf(data, (int)data_len);
-        x509 = d2i_X509_bio(input, NULL);
-        if (x509)
-        {
-            X509_print_ex_fp(
-                log_file, x509, XN_FLAG_COMPAT, XN_FLAG_SEP_CPLUS_SPC);
-        }
-        BIO_free_all(input);
-        fprintf(log_file, "\n");
+        X509_print_ex_fp(stdout, x509, XN_FLAG_COMPAT, XN_FLAG_SEP_CPLUS_SPC);
     }
+    BIO_free_all(input);
+    printf("\n");
 #endif
     OE_UNUSED(data);
     OE_UNUSED(data_len);
 }
 
-void validate_certificate(uint8_t* cert, size_t cert_size)
+oe_result_t validate_certificate(uint8_t* cert, size_t cert_size)
 {
     oe_result_t result;
 
     result = oe_verify_attestation_certificate(
         cert, cert_size, enclave_identity_verifier, NULL);
 
     log("Certificate verification result: %s\n", oe_result_str(result));
+
+    return result;
 }
 
-static oe_result_t _gen_cert(oe_enclave_t* enclave)
+oe_result_t get_report_from_cert(
+    const uint8_t* cert_in_der,
+    size_t cert_in_der_len,
+    uint8_t** report,
+    size_t* report_size)
+{
+    oe_result_t result = OE_OK;
+    uint8_t* report_buf = NULL;
+    size_t report_buf_size = cert_in_der_len;
+    oe_cert_t cert = {0};
+
+    result = oe_cert_read_der(&cert, cert_in_der, cert_in_der_len);
+    if (result != OE_OK)
+        return result;
+
+    report_buf = (uint8_t*)malloc(report_buf_size);
+    if (!report_buf)
+        return OE_OUT_OF_MEMORY;
+
+    // find the extension
+    result = oe_cert_find_extension(
+        &cert, X509_OID_FOR_QUOTE_STRING, report_buf, &report_buf_size);
+
+    if (result == OE_OK)
+    {
+        *report = report_buf;
+        *report_size = report_buf_size;
+    }
+    else
+    {
+        if (!report_buf)
+            free(report_buf);
+    }
+
+    return result;
+}
+
+oe_result_t gen_cert(oe_enclave_t* enclave, bool verbose)
 {
     oe_result_t result = OE_FAILURE;
     oe_result_t ecall_result;
     unsigned char* cert = NULL;
     size_t cert_size = 0;
```

Comment:
```suggestion
    unsigned char* certificate = nullptr;
    size_t certificate_size = 0;
```

(since this is a .cpp file, nullptr is preferred rather than NULL)